File: 암호만들기.py
def isValidPW(pw):
  mo=0 
  ja=0
  for ch in pw:
    if ch in 'aeiou': mo+=1
    else: ja+=1
  return mo >=1 and ja >=2

def go(index,alpha,pw,n):
  if len(pw)==n:
    if isValidPW(pw):
      print(pw)
    return
  if index == len(alpha) : return 
  go(index+1,alpha,pw+alpha[index],n)
  go(index+1,alpha,pw,n)

n,m = map(int,input().split())
alpha = input().split()
alpha.sort()
go(0,alpha,"",n)

File: 암호만들기.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

bool isValidPW(string &pw){
  int mo =0,ja=0;
  for (char ch: pw){
    if(ch=='a' || ch=='e'|| ch=='i'||ch=='o'||ch=='u') mo++;
    else ja++;
  }
  return ja>=2 && mo >=1;
}

void go(int index, vector<char>& alpha, string pw, int n ){
  if(pw.length() == n ) {
    if(isValidPW(pw)) cout<<pw<<'\n';
    return;
  }
  if(index == alpha.size() ) return;
  go(index+1,alpha,pw+alpha[index],n);
  go(index+1,alpha,pw,n);
}

int main(){
  int n,m;
  cin>>n>>m;
  vector<char> alpha(m);
  for(int i=0; i<m; i++){
    cin>>alpha[i];
  }
  sort(alpha.begin(),alpha.end());
  go(0,alpha,"",n);
}